From 85ad5dc9e0067cbb0bfa599f9c9d683d4c2da493 Mon Sep 17 00:00:00 2001
From: Tobias Jungel <tobias.jungel@bisdn.de>
Date: Mon, 8 Apr 2019 23:21:05 +0200
Subject: [PATCH 03/12] fix moar compiler warnings

builds/src/module/src/sysi.c:253:41: error: bitwise comparison always
evaluates to false [-Werror=tautological-compare]
|                      if((info.status&0x2)==1){
|                                          ^~

builds/src/module/src/sfpi.c:288:20: error: 'rx_los_all' may be used uninitialized in this function [-Werror=maybe-uninitialized]
|          rx_los_all |= status << (((i - 1)/ 8) * 8);
|          ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
---
 .../onlp/builds/src/module/src/sfpi.c         |  2 +-
 .../onlp/builds/src/module/src/sysi.c         | 86 +++++++++----------
 2 files changed, 44 insertions(+), 44 deletions(-)

diff --git a/packages/platforms/delta/x86-64/x86-64-delta-ag7648/onlp/builds/src/module/src/sfpi.c b/packages/platforms/delta/x86-64/x86-64-delta-ag7648/onlp/builds/src/module/src/sfpi.c
index 135be9e7..cef32f7d 100755
--- a/packages/platforms/delta/x86-64/x86-64-delta-ag7648/onlp/builds/src/module/src/sfpi.c
+++ b/packages/platforms/delta/x86-64/x86-64-delta-ag7648/onlp/builds/src/module/src/sfpi.c
@@ -262,7 +262,7 @@ onlp_sfpi_rx_los_bitmap_get(onlp_sfp_bitmap_t* dst)
 {
     int status;
     int   port,i = 0;
-	uint64_t rx_los_all;
+	uint64_t rx_los_all = 0;
 
 
 	if(platform_id == PLATFORM_ID_DELTA_AG7648_R0)
diff --git a/packages/platforms/delta/x86-64/x86-64-delta-ag7648/onlp/builds/src/module/src/sysi.c b/packages/platforms/delta/x86-64/x86-64-delta-ag7648/onlp/builds/src/module/src/sysi.c
index 7c4ca2ee..a0a4287b 100755
--- a/packages/platforms/delta/x86-64/x86-64-delta-ag7648/onlp/builds/src/module/src/sysi.c
+++ b/packages/platforms/delta/x86-64/x86-64-delta-ag7648/onlp/builds/src/module/src/sysi.c
@@ -43,18 +43,18 @@ platform_id_t platform_id = PLATFORM_ID_UNKNOWN;
 #define ONIE_PLATFORM_NAME "x86-64-delta-ag7648-r0"
 const char*
 onlp_sysi_platform_get(void)
-{ 
+{
 	enum ag7648_product_id pid = get_product_id();
-	
+
 	if (pid == PID_AG7648)
 		return "x86-64-delta-ag7648";
-	else 
+	else
 		return "unknow";
 }
 
 int
 onlp_sysi_platform_set(const char* platform)
-{ 
+{
     if(strstr(platform,"x86-64-delta-ag7648-r0")) {
         platform_id = PLATFORM_ID_DELTA_AG7648_R0;
         return ONLP_STATUS_OK;
@@ -65,11 +65,11 @@ onlp_sysi_platform_set(const char* platform)
 
 int
 onlp_sysi_platform_info_get(onlp_platform_info_t* pi)
-{ 
+{
     int   v;
-    
+
 	v = i2c_devname_read_byte("SYSCPLD", 0X0);
-	
+
     pi->cpld_versions = aim_fstrdup("%d", v & 0xf);
 
     return 0;
@@ -98,19 +98,19 @@ onlp_sysi_onie_data_get(uint8_t** data, int* size)
 			AIM_LOG_ERROR("Unable to read the %d reg \r\n",i);
 			break;
 		}
-			
+
 	}
-   
+
     *data = rdata;
 
     return ONLP_STATUS_OK;
 
-	
+
 }
 
 void
 onlp_sysi_onie_data_free(uint8_t* data)
-{ 
+{
     aim_free(data);
 }
 
@@ -118,7 +118,7 @@ onlp_sysi_onie_data_free(uint8_t* data)
 
 int
 onlp_sysi_oids_get(onlp_oid_t* table, int max)
-{ 
+{
     int i;
     onlp_oid_t* e = table;
     memset(table, 0, max*sizeof(onlp_oid_t));
@@ -147,7 +147,7 @@ onlp_sysi_oids_get(onlp_oid_t* table, int max)
 }
 int
 onlp_sysi_onie_info_get(onlp_onie_info_t* onie)
-{   
+{
     if(onie){
         onie->platform_name = aim_strdup(ONIE_PLATFORM_NAME);
     }
@@ -158,48 +158,48 @@ onlp_sysi_onie_info_get(onlp_onie_info_t* onie)
 
 int
 onlp_sysi_platform_manage_fans(void)
-{ 
+{
 
     int rc;
     onlp_thermal_info_t ti2, ti3, ti4;
     int mtemp=0;
     int new_rpm=0;
-    
+
     if (chassis_fan_count() == 0) {
         return ONLP_STATUS_E_UNSUPPORTED;
     }
-  
+
     /* Get temperature */
     /*rc = onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(1), &ti1);
-	
+
     if (rc != ONLP_STATUS_OK) {
         return rc;
     }*/
-        
+
     rc = onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(2), &ti2);
-	
+
     if (rc != ONLP_STATUS_OK) {
         return rc;
     }
     rc = onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(3), &ti3);
-	
+
     if (rc != ONLP_STATUS_OK) {
         return rc;
     }
-        
+
     rc = onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(4), &ti4);
-	
+
     if (rc != ONLP_STATUS_OK) {
         return rc;
     }
-             
+
     mtemp=(ti2.mcelsius+ti3.mcelsius + ti4.mcelsius) / 3;
 
     DEBUG_PRINT("mtemp %d\n", mtemp);
 
     /* Bring fan speed according the temp
      */
-  
+
     if(mtemp<25000)
         new_rpm=FAN_IDLE_RPM;
     else if((mtemp>=30000)&&(mtemp<40000))
@@ -213,25 +213,25 @@ onlp_sysi_platform_manage_fans(void)
     else{
         return ONLP_STATUS_OK;
    }
-  
+
     onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(1),new_rpm);
-    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(2),new_rpm); 
-    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(3),new_rpm); 
-    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(4),new_rpm); 
-    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(5),new_rpm); 
+    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(2),new_rpm);
+    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(3),new_rpm);
+    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(4),new_rpm);
+    onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(5),new_rpm);
     onlp_fani_rpm_set(ONLP_FAN_ID_CREATE(6),new_rpm);
 
 
-     
- 
-     
+
+
+
     return ONLP_STATUS_OK;
 }
 
 
 int
 onlp_sysi_platform_manage_leds(void)
-{ 
+{
 		int i,tray_i,rc;
 		onlp_fan_info_t info;
 		onlp_led_mode_t fan_new_mode;
@@ -250,7 +250,7 @@ onlp_sysi_platform_manage_leds(void)
                     goto tray_next;
                 }
                 else{
-                    if((info.status&0x2)==1){
+                    if(info.status&0x2){
                         fan_tray_new_mode[tray_i]=ONLP_LED_MODE_YELLOW;
                         goto tray_next;
                     }
@@ -262,21 +262,21 @@ tray_next:  continue;
         onlp_ledi_mode_set(ONLP_LED_ID_CREATE(LED_FAN_TRAY0),fan_tray_new_mode[0]);
         onlp_ledi_mode_set(ONLP_LED_ID_CREATE(LED_FAN_TRAY1),fan_tray_new_mode[1]);
         onlp_ledi_mode_set(ONLP_LED_ID_CREATE(LED_FAN_TRAY2),fan_tray_new_mode[2]);
-        
-        if((fan_tray_new_mode[0]==ONLP_LED_MODE_GREEN)&&(fan_tray_new_mode[1]==ONLP_LED_MODE_GREEN)&& 
+
+        if((fan_tray_new_mode[0]==ONLP_LED_MODE_GREEN)&&(fan_tray_new_mode[1]==ONLP_LED_MODE_GREEN)&&
             (fan_tray_new_mode[2]==ONLP_LED_MODE_GREEN))
             fan_new_mode=ONLP_LED_MODE_GREEN;
-        else if((fan_tray_new_mode[0]==ONLP_LED_MODE_OFF)||(fan_tray_new_mode[1]==ONLP_LED_MODE_OFF)|| 
+        else if((fan_tray_new_mode[0]==ONLP_LED_MODE_OFF)||(fan_tray_new_mode[1]==ONLP_LED_MODE_OFF)||
             (fan_tray_new_mode[2]==ONLP_LED_MODE_OFF))
              fan_new_mode=ONLP_LED_MODE_YELLOW;
         else
             fan_new_mode=ONLP_LED_MODE_YELLOW_BLINKING;
-    
+
 		onlp_ledi_mode_set(ONLP_LED_ID_CREATE(LED_FAN),fan_new_mode);
 		/*psu1 and psu2 led */
         for(i=1;i<=CHASSIS_PSU_COUNT;i++){
             rc=onlp_psui_info_get(ONLP_PSU_ID_CREATE(i),&psu);
-            
+
             if (rc != ONLP_STATUS_OK) {
                continue;
             }
@@ -286,17 +286,17 @@ tray_next:  continue;
             }
         }
 		psu_new_mode=ONLP_LED_MODE_YELLOW_BLINKING;
-     
-sys_led	:		
+
+sys_led	:
 		onlp_ledi_mode_set(ONLP_LED_ID_CREATE(LED_POWER),psu_new_mode);
 		//sys led 	----------------
 		if((fan_new_mode!=ONLP_LED_MODE_GREEN)||(psu_new_mode!=ONLP_LED_MODE_GREEN))
 			sys_new_mode=ONLP_LED_MODE_YELLOW_BLINKING;
 		else
 			sys_new_mode=ONLP_LED_MODE_GREEN;
-		
+
 		onlp_ledi_mode_set(ONLP_LED_ID_CREATE(LED_SYS),sys_new_mode);
-        
+
         locator_new_mode=ONLP_LED_MODE_GREEN;
         onlp_ledi_mode_set(ONLP_LED_ID_CREATE(LED_LOCATOR),locator_new_mode);
 		return ONLP_STATUS_OK;
-- 
2.20.1

